functions/time/gmtime.c

```diff
@@ -1,16 +1,111 @@
 /* gmtime( const time_t * )
-
-   This file is part of the Public Domain C Library (PDCLib).
-   Permission is granted to use, modify, and / or redistribute at will.
+ * From newlib gmtime_r.c
+ * Original Author: Adapted from tzcode maintained by Arthur David Olson.
+ *
+ *  This file is part of the Public Domain C Library (PDCLib).
+ *  Permission is granted to use, modify, and / or redistribute at will.
```

Comment:
This section is no longer true; copying from newlib is also problematic as it conflicts with the pdclib license and it's not clearly marked as such.

The file in newlib is actually problematic in itself because it doesn't have explicit copyright headers, but mentions a couple of contributors and external sources. Strictly speaking we'd have to add https://www.sourceware.org/git/gitweb.cgi?p=newlib-cygwin.git;a=blob;f=COPYING.NEWLIB and also ship it with each release of an nxdk app - this feels overkill to me.

I guess we could say that some of these algorithms are fact and non-copyrightable (as are the constants), but it still feels "bad".

Reply:
Well, there even is a pointer in the comments to a github page that describe the algorithm. Being what it is (converting seconds since 1970 to day/month/etc) I don't think there are so many ways to do it.